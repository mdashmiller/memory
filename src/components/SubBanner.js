import React from 'react'
import styled from 'styled-components'

const StyledSubBanner = styled.div`
	grid-column: 2 / span 2;
	/* border: 1px solid black; */
	height: 24px;
	text-align: center;
	font-size: 19px;
	font-weight: 700;
	color: #fccdd3;
`

const displayScore = score =>
	score ? score : '?'

const SubBanner = ({ bestScore }) =>
	<StyledSubBanner id="sub-banner">
		Best Score: {displayScore(bestScore)}
	</StyledSubBanner>

export default SubBanner
