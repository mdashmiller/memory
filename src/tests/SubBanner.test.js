import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import chai, { expect } from 'chai'

import SubBanner from '../components/SubBanner'

Enzyme.configure({ adapter: new Adapter() })

global.jestExpect = global.expect
global.expect = chai.expect

describe('<SubBanner />', () => {

	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<SubBanner />, div)
		ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(
			<SubBanner />
		)
		let tree = component.toJSON()
		jestExpect(tree).toMatchSnapshot()
	})

	it('renders 1 div #sub-banner', () => {
		const wrapper = shallow(<SubBanner />)
		expect(wrapper.find('#sub-banner')).to.have.lengthOf(1)
	})

	it('displays the best score when it exists', () => {
		const bestScore = 1
		const wrapper = shallow(<SubBanner bestScore={bestScore} />)
		expect(wrapper.find('#sub-banner').props().children[1])
			.to.equal(1)
	})

	it('displays a ? char when no best score exists', () => {
		const bestScore = null
		const wrapper = shallow(<SubBanner bestScore={bestScore} />)
		expect(wrapper.find('#sub-banner').props().children[1])
			.to.equal('?')
	})

})
