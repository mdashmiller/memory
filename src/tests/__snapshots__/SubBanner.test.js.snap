// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubBanner /> has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa lcbIRV"
  id="sub-banner"
>
  Best Score: 
  ?
</div>
`;
